src/haxeLanguageServer/features/haxe/codeAction/CodeActionFeature.hx

```diff
@@ -20,9 +26,11 @@ class CodeActionFeature {
 
 		context.registerCapability(CodeActionRequest.type, {
 			documentSelector: Context.haxeSelector,
-			codeActionKinds: [QuickFix, SourceOrganizeImports, SourceSortImports, RefactorExtract]
+			codeActionKinds: [QuickFix, SourceOrganizeImports, SourceSortImports, RefactorExtract],
+			resolveProvider: true,
```

Comment:
You mean making second `registerCapability` call for QuickFix specifically?
Not sure if there is much impact to allow resolve for all, resolve only triggers when there is no `command` and `edit` fields in action, as i understand.
I'm still thinking if i need to keep resolve at all.
Is there a way to run client command from server after code action, or will be better to move basic code action provider to promises? So action menu will have lag of fraction of second, but then it will be possible to run `command` after `edit` (to conver edits to snippets, autosave, etc)

Reply:
moving to promises would help for future expansion of code actions / refactoring options.
right now most code actions only come with minor edits that can be easily calculated on the spot with little overhead / delay.
once we add more refactoring options edit operations (or their calculation) can become quite complex. so having a way to tell vscode "you can do &lt;name of code action> here" without having to already have everything ready to go would be nice for these cases. and your feature already touches that kind of territory.

as to the `registerCapability`: I think `QuickFix` is not really fitting, `RefactorRewrite` might be a better match.

Reply:
I keeped `QuickFix` because there is no way to sort action kinds,  and `Add argument` after `Extract constant` in list, when you request actions for argument, seems annoying for me (i think error fixes like this should be upper than refactoring that just moves error to new pos).
Also i cannot call `registerCapability` twice to make `resolveProvider: true` specific for some kinds, there is action duplicates then.